src/main/java/com/sphy/hotelmanagementapplication/factory/ReverseModelMapperFactory.java

```diff
@@ -39,7 +45,7 @@ public ModelMapper create(ModelMapperFactory.ModelMapperType modelMapperType) {
 
                 // Custom mapping for Room, defining the way modelMapper maps the Hotel object to a Long
                 propertyMapper.addMappings(
-                        mapper -> mapper.map(src -> hotelRepository.findById(src.getHotel()), Room::setHotel)
+                        mapper -> mapper.map(src -> hotelRepository.findById(src.getHotel()).isPresent(), Room::setHotel)
```

Comment:
In the statement, byh using inside there the isPresent() method, you pass a boolean value, not the object itself.